src/pydvl/value/results.py

```diff
@@ -0,0 +1,272 @@
+
+    These include indices in the original :class:`Dataset`, any data names (e.g.
+    group names in :class:`GroupedDataset`), the values themselves, and standard
+    errors in the case of Monte Carlo methods. These can iterated over like any
+    ``Sequence``: ``iter(valuation_result)`` returns a generator of
+    :class:`ValueItem` in the order in which the object is sorted.
+
+    Results can be sorted in-place with :meth:`sort` or using python's standard
+    ``sorted()`` and ``reversed()`` Note that sorting values affects how
+    iterators and the object itself as ``Sequence`` behave: ``values[0]``
+    returns a :class:`ValueItem` with the highest or lowest ranking point if
+    this object is sorted by descending or ascending value, respectively. If
+    unsorted, ``values[0]`` returns a ``ValueItem`` for index 0.
+
+    :param algorithm: The method used.
+    :param status: The end status of the algorithm.
+    :param values: An array of values, data indices correspond to positions in
+        the array.
+    :param stderr: An optional array of standard errors in the computation of
+        each value.
+    :param data_names: Names for the data points. Defaults to index numbers
+        if not set.
+    :param sort: Whether to sort the indices by ascending value. See above how
+        this affects usage as an iterable or sequence.
+
+    :raise ValueError: If data names and values have mismatching lengths.
+    """
+
+    _indices: "NDArray[np.int_]"
+    _values: "NDArray[np.float_]"
+    _data: Dataset
+    _names: "Union[NDArray[np.int_], NDArray[np.str_]]"
+    _stderr: "NDArray[np.float_]"
+    _algorithm: str  # TODO: BaseValuator
+    _status: ValuationStatus  # TODO: Maybe? BaseValuator.Status
+    # None for unsorted, True for ascending, False for descending
+    _sort_order: Optional[bool]
+
+    def __init__(
+        self,
+        algorithm: str,  # BaseValuator,
+        status: ValuationStatus,  # Valuation.Status,
+        values: "NDArray[np.float_]",
+        stderr: Optional["NDArray[np.float_]"] = None,
+        data_names: Optional[Sequence[str]] = None,
+        sort: bool = True,
+    ):
+        if stderr is not None and len(stderr) != len(values):
+            raise ValueError("Lengths of values and stderr do not match")
+
+        self._algorithm = algorithm
+        self._status = status
+        self._values = values
+        self._stderr = np.zeros_like(values) if stderr is None else stderr
+        self._sort_order = None
+
+        if sort:
+            self.sort()
+        else:
+            self._indices = np.arange(0, len(self._values), dtype=np.int_)
+
+        if data_names is None:
+            self._names = np.arange(0, len(values), dtype=np.int_)
+        else:
+            self._names = np.array(data_names)
+        if len(self._names) != len(self._values):
+            raise ValueError("Data names and data values have different lengths")
+
+    def sort(self, reverse: bool = False) -> None:
```

Comment:
It would be nice to offer this interface instead to allow using the std in the comparison, for example
```suggestion
    def sort(self, *, key=None, reverse=False) -> None:
```
But this would mean that we cannot use argsort, which would make this slower.

Reply:
What happened to YAGNI? The moment we need to sort by stderr we can add this, ok?